File: JavaPatternPrograms/SpiralPatterns/Pattern 4.java
class SpiralNumberPattern2
{
    public static void main(String args[])
    {
        int i=1;
        int rows = 10;
        int k = 1;
        int x = 1;

        for (x = 1; x <= rows * 2; x += 2)
        {
            if (k % 2 == 1)
            {
                System.out.printf("%02d %02d %02d %02d ",i,(i+1),(i+2),(i+3));
                k++;
                i += 4;
            }
            else
            {
                System.out.printf("%02d %02d %02d %02d ",(i+3),(i+2),(i+1),i);
                k++;
                i += 4;
            }
            System.out.println();
        }
    }
}

File: PythonPatternPrograms/SpiralPatterns/Pattern 4.py
i = 1
rows = 10
k = 1

for x in range(1, rows * 2 + 1, +2):
    if k % 2 == 1:
        print("{:02d} ".format(i), end="")
        print("{:02d} ".format(i + 1), end="")
        print("{:02d} ".format(i + 2), end="")
        print("{:02d} ".format(i + 3), end="")
        k += 1
        i += 4
    else:
        print("{:02d} ".format(i + 3), end="")
        print("{:02d} ".format(i + 2), end="")
        print("{:02d} ".format(i + 1), end="")
        print("{:02d} ".format(i), end="")
        k += 1
        i += 4
    print()